src/lib.rs

```diff
@@ -58,6 +58,8 @@
 #![cfg_attr(feature = "nightly", feature(alloc))]
 #![cfg_attr(not(test), no_std)]
 
+#![deny(missing_docs, warnings, missing_debug_implementations)]
```

Comment:
I don't feel comfortable with `deny`ing lints because future compiler updates could suddenly make our code stop compiling. See https://github.com/rust-lang/rust/pull/47204#issuecomment-356608522 for a recent example of a new warning breaking existing crates. This can be especially frustrating for crates depending on crossbeam.

We should be fine with just `#![warn(missing_docs, missing_debug_implementations)]`.

And if that is not enough, we can add `RUSTFLAGS="-D warnings"` to the Travis config. Better to break our CI builds than the whole ecosystem. :)

Reply:
> This can be especially frustrating for crates depending on crossbeam.

AFAIA cargo caps lints when building dependencies, so it would only affect crossbeam itself.